src/commands/hld/reconcile.ts

```diff
@@ -101,6 +102,39 @@ type MiddlewareMap<T = Partial<ReturnType<typeof middleware.create>>> = {
   default?: T;
 };
 
+// In spk hld reconcile, the results should always result in the same artifacts being created based on the state of bedrock.yaml.
+// The only exception is for files under the /config directories and any access.yaml files.
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const purgeRepositoryComponents = (
+  absHldPath: string,
+  repositoryName: string
+): void => {
+  assertIsStringWithContent(absHldPath, "hld-path");
+  assertIsStringWithContent(repositoryName, "repository-name");
+
+  const filesToDelete = getAllFilesInDirectory(
+    path.join(absHldPath, repositoryName)
+  ).filter(
+    (filePath) =>
+      !filePath.match(/access\.yaml$/) && !filePath.match(/config\/.*\.yaml$/)
+  );
+
+  try {
+    filesToDelete.forEach((file) => {
+      fs.unlink(file, function (err) {
+        if (err) throw err;
+        console.log(`${file} deleted!`);
+      });
+    });
+  } catch (e) {
+    logger.error(
```

Comment:
Use buildError to create the error and logError to log it. Throw the buildError.

Reply:
replace 
```
logger.error(`error purging repository component '${repositoryName}' in path '${absHldPath}'`);
throw e;
```
with
```
      throw buildError(
        errorStatusCode.FILE_IO_ERR,
        {
          errorKey: "hld-reconcile-err-purge-repo-comps",
          values: [repositoryName, absHldPath],
        },
        err
      );
```
and add to `lib/i18n.json`

```
   "hld-reconcile-err-purge-repo-comps": "Could not purge repository component {0} in path {1}."
```

when the error happen, we will get
```
code: 1011
message: Could not purge repository component <something> in path <some path>.
details: <error message from err>
```

Reply:
You are correct, we can just throw the buildError, no need to log it in this case.